tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,226 @@
-
-        # Compute the c2st and assert it is near chance level of 0.5.
-        check_c2st(
-            samples,
-            target_samples,
-            alg=f"npse-{sde_type or 'vp'}-{prior_str}-{num_dim}D-{method}",
-        )
-
-    # Checks for log_prob()
-    if prior_str == "gaussian":
-        # For the Gaussian prior, we compute the KLd between ground truth and
-        # posterior.
-        dkl = get_dkl_gaussian_prior(
-            posterior,
-            x_o[0],
-            likelihood_shift,
-            likelihood_cov,
-            prior_mean,
-            prior_cov,
-        )
-
-        max_dkl = 0.15
-
-        assert dkl < max_dkl, (
-            f"D-KL={dkl} is more than 2 stds above the average performance."
-        )
-
-
-def test_c2st_npse_on_linearGaussian_different_dims():
-    """Test NPE on linear Gaussian with different theta and x dimensionality."""
-
-    theta_dim = 3
-    x_dim = 2
-    discard_dims = theta_dim - x_dim
-
-    x_o = zeros(1, x_dim)
-    num_samples = 1000
-    num_simulations = 2000
-
-    # likelihood_mean will be likelihood_shift+theta
-    likelihood_shift = -1.0 * ones(x_dim)
-    likelihood_cov = 0.3 * eye(x_dim)
-
-    prior_mean = zeros(theta_dim)
-    prior_cov = eye(theta_dim)
-    prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    target_samples = samples_true_posterior_linear_gaussian_mvn_prior_different_dims(
-        x_o,
-        likelihood_shift,
-        likelihood_cov,
-        prior_mean,
-        prior_cov,
-        num_discarded_dims=discard_dims,
-        num_samples=num_samples,
-    )
-
-    def simulator(theta):
-        return linear_gaussian(
-            theta,
-            likelihood_shift,
-            likelihood_cov,
-            num_discarded_dims=discard_dims,
+    theta, x = test_case.get_training_data(num_simulations)
+    if stop_after_epochs is None and training_batch_size is None:
+        score_estimator = inference.train()
+    elif stop_after_epochs is not None and training_batch_size is None:
+        score_estimator = inference.train(stop_after_epochs=stop_after_epochs)
+    elif stop_after_epochs is None and training_batch_size is not None:
+        score_estimator = inference.train(training_batch_size=training_batch_size)
+    else:
+        score_estimator = inference.append_simulations(theta, x).train(
+            stop_after_epochs=stop_after_epochs, training_batch_size=training_batch_size
         )
+    return inference, score_estimator
 
-    # Test whether prior can be `None`.
-    inference = NPSE(prior=None)
 
-    theta = prior.sample((num_simulations,))
-    x = simulator(theta)
-
-    # Test whether we can stop and resume.
-    inference.append_simulations(theta, x).train(
-        max_num_epochs=10, training_batch_size=100
+@pytest.fixture
+def npse_trained_model(request):
+    num_simulations = 5_000
+    stop_after_epochs = 200
+    training_batch_size = 100
+    test_case: NpseTestCase = request.param
```

Comment:
how do I know what's in the param?

Reply:
test_case is passed indirectly with `parametrize` before each test
`@pytest.mark.parametrize("npse_trained_model", test_cases_all, indirect=True)`